classes/Blocks.php

```diff
@@ -38,15 +38,54 @@ function register_blocks()
         );
     }
 
+    function enqueue_cardlink_events() {
+        wp_enqueue_script('cardlink-events', plugin_dir_url(__FILE__) . '../scripts/cardlink-events.js');
+    }
+
+    function enqueue_cardviewer_select_events() {
+        wp_enqueue_script('cardviewer-select-events', plugin_dir_url(__FILE__) . '../scripts/cardviewer-select-events.js');
+    }
+
     function bye_cardviewer_card_render($block_attributes, $content)
     {
         try {
-            if (array_key_exists('cardOfTheDay', $block_attributes) && $block_attributes['cardOfTheDay']) {
-                $carddata  =  $this->database->find_card_ofTheDay($block_attributes['language'] ?? 'en');
+            /*
+             * TODO: Add an optional frontend card selection for use in the card viewer
+             * > Card (Expansion + Name)/Version/Language dropdowns can be toggled separately in block config
+             * >> Expansion has fixed options [DONE]
+             * >> Name has cards in expansion [DONE]
+             * >> Version only what exists for this card [DONE]
+             * >> Language only what exists for this card [DONE]
+             * > The actual card displayed is the first of the following that applies [DONE]
+             * >> 1) The card selected in the dropdowns
+             * >> 2) The card given in the URL parameters
+             * >> 3) The card of the day (if enabled) => this allows us to use the generic card viewer as CotD!
+             * >> 4) The card configured in the backend
+             * > If dropdowns are present AND an initial card is present (via methods 2/3/4),
+             *   the dropdown values are also initialized appropriately [DONE]
+             * > If the displayed card is card of the day, regardless of the method it was selected by,
+             *   display "🎉 Card of the Day!" (or similar) somewhere [DONE]
+             */
+
+            $wrapper_attr = [];
+            $cotd = $this->database->find_card_ofTheDay($block_attributes['language'] ?? 'en');
+            if (array_key_exists('fromUrlParams', $block_attributes) && $block_attributes['fromUrlParams']) {
```

Comment:
For the sake of completeness, we should also allow passing language this way.

Reply:
Done. Also, $cotd is now only initialized after URL params are parsed, so we can use them to switch to a localized card of the day.